docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+rules of the sidechain system. Assume the funds are lost (the only way to
+recover them would be if the witness servers created a transaction themselves.
+But this is unlikely to happen and should not be relied upon.) The "Minimum
+account create" amount is meant to prevent these transactions from failing. If
+this transaction is used to try to create an already existing account, the funds
+are lost.
+
+## Trust Assumptions
+
+The witness servers are trusted, and if a quorum of them collude they can steal
+funds from the door account.
+
+## STSidechain
+
+## Ledger Objects
+
+Many of the ledger objects and transactions contain a `STSidechain` object. These are the parameters that
+define a sidechain. It contains the following fields:
+
+* srcChainDoor: `AccountID` of the door account on the mainchain. This account
+  will hold assets in trust while they are used on the sidechain.
+    
+* srcChainIssue: `Issue` of the asset put into trust on the mainchain.
+
+* dstChainDoor: `AccountID` of the door account on the sidechain. This account
+  will issue wrapped assets representing assets put into trust on the mainchain.
+    
+* dstChainIssue: `Issue` of the asset used to represent from the mainchain.
+
+Note: `srcChainDoor` and `dstChainDoor` must be distinct accounts. This is done
+    to help prevent transaction replay attacks.
+
+Note: `srcChainIssue` and `dstChainIssue` must both XRP or both be IOUs. This is
+    done because the exchange rate is fixed at 1:1, and IOUs and XRP have a
+    different numeric range and precision. This requirement may be relaxed in
+    the future.
+
+
+A snippet of the data for C++ class for an `STSidechain` is:
+
+```c++
+class STSidechain final : public STBase
+{
+    AccountID srcChainDoor_{};
+    Issue srcChainIssue_{};
+    AccountID dstChainDoor_{};
+    Issue dstChainIssue_{};
+}
+```
+
+### Sidechain
+
+The sidechain ledger object is owned by the door account and defines the
+sidechain parameters. Note, the signatures used to attest to chain events are on
+the door account, not on this ledger object. It is created with a
+`SidechainCreate` transaction, modified with a `SidechainModify` transaction
+(only the `MinAccountCreateAmount` and `SignaturesReward` may be changed). It
+can not be deleted.
+
+#### Fields
+
+The ledger object contains the following fields:
+
+* Account: The account that owns this object. The door account. Required.
+
+* Balance: If positive, the funds in held in trust on this door account. If
+  negative, the wrapped funds issued from this door account. Required.
+
+* SignaturesReward: Total amount, in XRP, to be rewarded for providing a
+  signatures for a cross-chain transfer or for signing for the cross-chain
+  reward. This will be split among the signers. Required.
+
+* SignatureRewardsBalance: Amount, in XRP, currently lock for rewarding signers.
+  This is paid by the accounts sending `SidechainXChainTransfer` transactions.
+  Required.
```

Comment:
naming nit: it feels confusing and really easy to typo that one is `SignaturesReward` and the other is `SignatureRewards`

Reply:
Yep, good comment. Will try to think of a better name.